### PR TITLE
change: use sagemaker-containers' ProcessRunner for executing entry p…

### PR DESCRIPTION
**Pull request by wiltonwu**
*Issue #, if available:*
aws/sagemaker-python-sdk#941

*Description of changes:*
* this change is specifically to take advantage of https://github.com/aws/sagemaker-containers/blob/master/src/sagemaker_containers/_process.py#L76-L86
* same as aws/sagemaker-pytorch-container#135 and aws/sagemaker-mxnet-container#114
* the approach is copied from https://github.com/aws/sagemaker-chainer-container/blob/master/src/sagemaker_chainer_container/training.py (but without the MPI stuff)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ericangelokim** [comment]
Tests run? Also was a notebook run?

File: src/sagemaker_sklearn_container/training.py [hunk 1]
```diff
@@ -28,9 +28,11 @@ def train(training_environment):
                                training arguments and hyperparameters
     """
     logger.info('Invoking user training script.')
-    framework.modules.run_module(training_environment.module_dir, training_environment.to_cmd_args(),
-                                 training_environment.to_env_vars(), training_environment.module_name)
+    framework.modules.download_and_install(training_environment.module_dir)
+    framework.entry_point.run(training_environment.module_dir, training_environment.user_entry_point,
+                              training_environment.to_cmd_args(), training_environment.to_env_vars(),
+                              runner=framework.runner.ProcessRunnerType)
```
Comment (ericangelokim) [inline src/sagemaker_sklearn_container/training.py]: This is the default runner already right? Was this included for awareness?

Comment (wiltonwu) [reply src/sagemaker_sklearn_container/training.py]: Yes

**Comment by wiltonwu** [comment]
Confirmed that pipeline tests and notebook with new image both work

**Comment by adurivault** [comment]
I seems to me that this did not fix the problem. I created a github repo to reproduce the problem. When I run the following code from a sagemaker notebook : 

```
from sagemaker.sklearn import SKLearn
import sagemaker

git_config = {
    'repo': 'https://github.com/adurivault/aws-sagemaker-example.git',
    'branch': 'master',
}

skl_estimator = SKLearn(
    'src/main.py',
    train_instance_type="local",
    role=sagemaker.get_execution_role(),
    source_dir='sub-dir',
    git_config=git_config,
)

skl_estimator.fit({})
```

I get the following error : 

```
Invoking script with the following command:

/miniconda3/bin/python -m src/main


/miniconda3/bin/python: No module named src/main
2019-11-28 11:05:22,748 sagemaker-containers ERROR    ExecuteUserScriptError:
Command "/miniconda3/bin/python -m src/main"
```

It seems to me that the correct command  should be `/miniconda3/bin/python -m src.main`

Am I missing something ?

**Comment by iyerr3** [comment]
Try:

skl_estimator = SKLearn(
    'main.py',
    train_instance_type="local",
    role=sagemaker.get_execution_role(),
    source_dir='sub-dir/src',
    git_config=git_config,
)

I was able to move past the error you were seeing (though didn’t complete training as I don’t have access to your s3 bucket).

Best,
Rahul

**Comment by adurivault** [comment]
Yes, this does indeed solve the issue and this is what I've been using. 
But this changes the root directory of my project, and therefore the way I have to deal with imports. So I still think this is an issue that needs to be fixed, and that this solution is just a workaround.

**Comment by iyerr3** [comment]
Makes sense - can you please move this to an issue for us to track?
